lib/compilers/dotnet.ts

```diff
@@ -283,46 +283,69 @@ class DotNetCompiler extends BaseCompiler {
         inputFilename: string,
         execOptions: ExecutionOptions & {env: Record<string, string>},
     ): Promise<CompilationResult> {
-        let toolOptions: string[] = [];
-        const toolSwitches: string[] = [];
-        const configurableOptions = this.configurableOptions;
+        const corerunArgs: string[] = [];
+        const ilcOptions: string[] = [
+            '--codegenopt',
+            'JitDisasm=*',
+            '--codegenopt',
+            'JitDisasmDiffable=1',
+            '--parallelism',
+            '1',
+        ];
+        const ilcSwitches: string[] = [];
         const programDir = path.dirname(inputFilename);
         const programOutputPath = path.join(programDir, 'bin', this.buildConfig, this.targetFramework);
-        const programDllPath = path.join(programOutputPath, `${AssemblyName}.dll`);
-
-        for (const configurableOption of configurableOptions) {
-            const optionIndex = options.indexOf(configurableOption);
-            if (optionIndex === -1 || optionIndex === options.length - 1) {
-                continue;
-            }
-            toolOptions.push(options[optionIndex], options[optionIndex + 1]);
-        }
-
+        const programDllPath = path.join(programOutputPath, 'CompilerExplorer.dll');
+        const envVarFileContents = [
+            'DOTNET_EnableWriteXorExecute=0',
+            'DOTNET_JitDisasm=*',
+            'DOTNET_JitDisasmAssemblies=CompilerExplorer',
+            'DOTNET_TieredCompilation=0',
+        ];
         let isAot = false;
-        const configurableSwitches = this.configurableSwitches;
-        for (const configurableSwitch of configurableSwitches) {
-            const switchIndex = options.indexOf(configurableSwitch);
-            if (switchIndex === -1) {
+
+        while (options.length > 0) {
```

Comment:
Just so I'm reading this...this is parsing the user's cmdline options, right? We've avoided this before as there's usually a half-dozen ways to specify cmdline args, but your code here appears to handle most. Can you use the `--env=blah` type syntax for these args?

Reply:
This is because corerun doesn't support passing environment variables using `-e name=value`. Instead we need to parse all environment variables manually write them to a file, and pass `-e file` to corerun.